ext/OptimizationEnzymeExt.jl

```diff
@@ -83,22 +73,37 @@ function lagrangian(x, _f::Function, cons::Function, p, λ, σ = one(eltype(x)))
     return σ * _f(x, p) + dot(λ, res)
 end
 
-function lag_grad(x, dx, lagrangian::Function, _f::Function, cons::Function, p, σ, λ)
+function lag_grad(mode, x, dx, lagrangian::Function, _f::Function, cons::Function, p, σ, λ)
     Enzyme.autodiff_deferred(
-        Enzyme.Reverse, Const(lagrangian), Active, Enzyme.Duplicated(x, dx),
+        mode, Const(lagrangian), Active, Enzyme.Duplicated(x, dx),
         Const(_f), Const(cons), Const(p), Const(λ), Const(σ))
     return nothing
 end
 
+
+set_runtime_activity2(a::Mode1, ::Enzyme.Mode{ABI, Err, RTA}) where {Mode1, ABI, Err, RTA} = Enzyme.set_runtime_activity(a, RTA)
```

Comment:
@wsmoses I am not sure what you wanted to do here, it can probably be hacked together with `supertype(typeof(a))` or similar but my guess is you expect this to be available more easily here?

Reply:
Surprisingly it compiles here but fails for me locally to precompile because of this

Reply:
Bump @wsmoses

Reply:
@Vaibhavdixit02 so we need to take the activity from the input specification, if requested, and use it to set the activity in the Forward/Reverse mode's Optimization.jl passes. How does it fail to precompile for you offhand?

Reply:
ah, it should be like `set_runtime_activity2(Enzyme.Forward, adtype.mode)` and `set_runtime_activity2(Enzyme.Reverse, adtype.mode)` `